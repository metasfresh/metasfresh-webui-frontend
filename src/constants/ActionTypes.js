export * from './TableTypes';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const NO_CONNECTION = 'NO_CONNECTION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const OPEN_MODAL = 'OPEN_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLOSE_MODAL = 'CLOSE_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLOSE_PROCESS_MODAL = 'CLOSE_PROCESS_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_MODAL = 'UPDATE_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const USER_SESSION_INIT = 'USER_SESSION_INIT';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const USER_SESSION_UPDATE = 'USER_SESSION_UPDATE';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SET_PROCESS_STATE_SAVED = 'SET_PROCESS_STATE_SAVED';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SET_PROCESS_STATE_PENDING = 'SET_PROCESS_STATE_PENDING';

// SCOPED ACTIONS

/**
 * @constant
 * Window is being initialized, data requests are triggered
 * @type {string}
 */
export const INIT_WINDOW = 'INIT_WINDOW';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const INIT_DATA_SUCCESS = 'INIT_DATA_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const INIT_LAYOUT_SUCCESS = 'INIT_LAYOUT_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ADD_ROW_DATA = 'ADD_ROW_DATA';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SORT_TAB = 'SORT_TAB';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ACTIVATE_TAB = 'ACTIVATE_TAB';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UNSELECT_TAB = 'UNSELECT_TAB';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ADD_NEW_ROW = 'ADD_NEW_ROW';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DELETE_ROW = 'DELETE_ROW';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_ROW_PROPERTY = 'UPDATE_ROW_PROPERTY';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_DATA_PROPERTY = 'UPDATE_DATA_PROPERTY';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_DATA_FIELD_PROPERTY = 'UPDATE_DATA_FIELD_PROPERTY';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_ROW_FIELD_PROPERTY = 'UPDATE_ROW_FIELD_PROPERTY';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_ROW_STATUS = 'UPDATE_ROW_STATUS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_MASTER_DATA = 'UPDATE_MASTER_DATA';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLEAR_MASTER_DATA = 'CLEAR_MASTER_DATA';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_DATA_VALID_STATUS = 'UPDATE_DATA_VALID_STATUS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_DATA_SAVE_STATUS = 'UPDATE_DATA_SAVE_STATUS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_DATA_INCLUDED_TABS_INFO = 'UPDATE_DATA_INCLUDED_TABS_INFO';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ADD_VIEW_LOCATION_DATA = 'ADD_VIEW_LOCATION_DATA';

/**
 * @constant
 * @type {string} Delete view from redux state
 */
export const DELETE_VIEW = 'DELETE_VIEW';

/**
 * @constant
 * @type {string} Reset view state in redux tree
 */
export const RESET_VIEW = 'RESET_VIEW';

/**
 * @constant
 * @type {string} Fetch data request
 */
export const FETCH_DOCUMENT_PENDING = 'FETCH_DOCUMENT_PENDING';

/**
 * @constant
 * @type {string} Success fetching data
 */
export const FETCH_DOCUMENT_SUCCESS = 'FETCH_DOCUMENT_SUCCESS';

/**
 * @constant
 * @type {string} Error when fetching data
 */
export const FETCH_DOCUMENT_ERROR = 'FETCH_DOCUMENT_ERROR';

/**
 * @constant
 * @type {string} Trigger fetching layout request
 */
export const FETCH_LAYOUT_PENDING = 'FETCH_LAYOUT_PENDING';

/**
 * @constant
 * @type {string} Success when fetching view layout
 */
export const FETCH_LAYOUT_SUCCESS = 'FETCH_LAYOUT_SUCCESS';

/**
 * @constant
 * @type {string} Error when fetching view layout
 */
export const FETCH_LAYOUT_ERROR = 'FETCH_LAYOUT_ERROR';

/**
 * @constant
 * @type {string} Trigger creating new view
 */
export const CREATE_VIEW = 'CREATE_VIEW';

/**
 * @constant
 * @type {string} Success when creating new view
 */
export const CREATE_VIEW_SUCCESS = 'CREATE_VIEW_SUCCESS';

/**
 * @constant
 * @type {string} Error when creating new view
 */
export const CREATE_VIEW_ERROR = 'CREATE_VIEW_ERROR';

/**
 * @constant
 * @type {string} Trigger filtering view
 */
export const FILTER_VIEW_PENDING = 'FILTER_VIEW_PENDING';

/**
 * @constant
 * @type {string} Success when filtering view
 */
export const FILTER_VIEW_SUCCESS = 'FILTER_VIEW_SUCCESS';

/**
 * @constant
 * @type {string} Error when filtering view
 */
export const FILTER_VIEW_ERROR = 'FILTER_VIEW_ERROR';

/**
 * @constant
 * @type {string} Update list data
 */
export const UPDATE_VIEW_DATA = 'UPDATE_VIEW_DATA';

/**
 * @constant
 * @type {string} Success when getting location config
 */
export const FETCH_LOCATION_CONFIG_SUCCESS = 'FETCH_LOCATION_CONFIG_SUCCESS';

/**
 * @constant
 * @type {string} Error when getting location config
 */
export const FETCH_LOCATION_CONFIG_ERROR = 'FETCH_LOCATION_CONFIG_ERROR';

// END OF SCOPED ACTIONS

// INDICATOR ACTIONS
/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CHANGE_INDICATOR_STATE = 'CHANGE_INDICATOR_STATE';
// END OF INDICATOR ACTIONS

// NOTIFICATION ACTIONS
/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SET_NOTIFICATION_PROGRESS = 'SET_NOTIFICATION_PROGRESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DELETE_NOTIFICATION = 'DELETE_NOTIFICATION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLEAR_NOTIFICATIONS = 'CLEAR_NOTIFICATIONS';
// END OF NOTIFICATION ACTIONS

//AUTH ACTIONS
/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ENABLE_TUTORIAL = 'ENABLE_TUTORIAL';
//END OF AUTH ACTIONS

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const INIT_FILTERS_PARAMETERS = 'INIT_FILTERS_PARAMETERS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DELETE_FILTERS_PARAMETERS = 'DELETE_FILTERS_PARAMETERS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_FILTERS_PARAMETERS = 'UPDATE_FILTERS_PARAMETERS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DRAGGABLE_CARD = 'DRAGGABLE_CARD';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const NEW_PROCESS_SUCCESS = 'NEW_PROCESS_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const REMOVE_TABLE_ITEMS_SELECTION = 'REMOVE_TABLE_ITEMS_SELECTION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SELECT_TABLE_ITEMS = 'SELECT_TABLE_ITEMS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DESELECT_TABLE_ITEMS = 'DESELECT_TABLE_ITEMS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const GET_NOTIFICATIONS_SUCCESS = 'GET_NOTIFICATIONS_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const NEW_NOTIFICATION = 'NEW_NOTIFICATION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const REMOVE_NOTIFICATION = 'REMOVE_NOTIFICATION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const REMOVE_ALL_NOTIFICATIONS = 'REMOVE_ALL_NOTIFICATIONS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const READ_NOTIFICATION = 'READ_NOTIFICATION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const READ_ALL_NOTIFICATIONS = 'READ_ALL_NOTIFICATIONS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const OPEN_PLUGIN_MODAL = 'OPEN_PLUGIN_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLOSE_PLUGIN_MODAL = 'CLOSE_PLUGIN_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const OPEN_RAW_MODAL = 'OPEN_RAW_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_RAW_MODAL = 'UPDATE_RAW_MODAL';

/**
 * @constant
 * @type {string}
 */
export const SET_RAW_MODAL_TITLE = 'SET_RAW_MODAL_TITLE';

/**
 * @constant
 * @type {string}
 */
export const SET_RAW_MODAL_DESCRIPTION = 'SET_RAW_MODAL_DESCRIPTION';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLOSE_RAW_MODAL = 'CLOSE_RAW_MODAL';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const TOGGLE_OVERLAY = 'TOGGLE_OVERLAY';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const OPEN_FILTER_BOX = 'OPEN_FILTER_BOX';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const CLOSE_FILTER_BOX = 'CLOSE_FILTER_BOX';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SET_LATEST_NEW_DOCUMENT = 'SET_LATEST_NEW_DOCUMENT';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ALLOW_SHORTCUT = 'ALLOW_SHORTCUT';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DISABLE_SHORTCUT = 'DISABLE_SHORTCUT';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const PATCH_REQUEST = 'PATCH_REQUEST';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const PATCH_SUCCESS = 'PATCH_SUCCESS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const PATCH_FAILURE = 'PATCH_FAILURE';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const PATCH_RESET = 'PATCH_RESET';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const ALLOW_OUTSIDE_CLICK = 'ALLOW_OUTSIDE_CLICK';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const DISABLE_OUTSIDE_CLICK = 'DISABLE_OUTSIDE_CLICK';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SHOW_SPINNER = 'SHOW_SPINNER';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const HIDE_SPINNER = 'HIDE_SPINNER';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const INIT_KEYMAP = 'INIT_KEYMAP';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_KEYMAP = 'UPDATE_KEYMAP';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const INIT_HOTKEYS = 'INIT_HOTKEYS';

/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const UPDATE_HOTKEYS = 'UPDATE_HOTKEYS';

/**
 * @constant
 * @type {string} fetches panel's quick actions.
 */
export const FETCHED_QUICK_ACTIONS = 'FETCHED_QUICK_ACTIONS';

/**
 * @constant
 * @type {string} removes panel's quick actions when it's unmounted.
 */
export const DELETE_QUICK_ACTIONS = 'DELETE_QUICK_ACTIONS';

/**
 * @constant
 * @type {string} fetches tab's quick actions.
 */
export const FETCH_TOP_ACTIONS = 'FETCH_TOP_ACTIONS';

/**
 * @constant
 * @type {string} called on successful response.
 */
export const FETCH_TOP_ACTIONS_SUCCESS = 'FETCH_TOP_ACTIONS_SUCCESS';

/**
 * @constant
 * @type {string} called on failed response.
 */
export const FETCH_TOP_ACTIONS_FAILURE = 'FETCH_TOP_ACTIONS_FAILURE';

/**
 * @constant
 * @type {string} removes tab's quick actions on unmount.
 */
export const DELETE_TOP_ACTIONS = 'DELETE_TOP_ACTIONS';

/**
 * @constant
 * @type {string} removes tab's quick actions on unmount.
 */
export const UPDATE_TAB_ROWS_DATA = 'UPDATE_TAB_ROWS_DATA';

/**
 * @constant
 * @type {string} set languages in the app handler
 */
export const SET_LANGUAGES = 'SET_LANGUAGES';
