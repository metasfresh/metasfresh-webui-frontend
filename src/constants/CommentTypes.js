// COMMENTS ENTITY ACTIONS
export const UPDATE_COMMENTS_PANEL = 'UPDATE_COMMENTS_PANEL';
export const UPDATE_COMMENTS_PANEL_TEXT_INPUT =
  'UPDATE_COMMENTS_PANEL_TEXT_INPUT';
// END OF COMMENT ACTIONS
