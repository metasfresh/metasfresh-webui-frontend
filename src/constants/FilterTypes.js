// FILTTER ACTIONS
export const SET_CLEAR_ALL_FILTER = 'SET_CLEAR_ALL_FILTER';
// END OF FILTER ACTIONS
