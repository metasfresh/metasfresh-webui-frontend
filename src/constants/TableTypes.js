/**
 * @constant
 * @type {string} ToDo: Description for the constant.
 */
export const SET_ACTIVE_SORT = 'SET_ACTIVE_SORT';

/**
 * @constant
 * @type {string} Init new table.
 */
export const CREATE_TABLE = 'CREATE_TABLE';

/**
 * @constant
 * @type {string} Update table data.
 */
export const UPDATE_TABLE = 'UPDATE_TABLE';
