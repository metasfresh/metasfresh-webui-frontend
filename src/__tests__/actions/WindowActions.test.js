import thunk from 'redux-thunk'
import nock from 'nock';
import configureStore from 'redux-mock-store';

import {
  updateTabRowsData,
  initDataSuccess,
  initLayoutSuccess,
  fetchTopActions,
  createWindow,
  initWindow,
} from '../../actions/WindowActions';
import * as ACTION_TYPES from '../../constants/ActionTypes';

import fixtures from "../../../test_setup/fixtures/master_window.json";

describe('WindowActions synchronous', () => {
  it('should return a UPDATE_TAB_ROWS_DATA action with correct payload', () => {
    const payload = { 'AD_Tab-1': { changed: { id: 1 }, removed: {} }};
    const action = updateTabRowsData('master', 'AD_Tab-1', payload);

    expect(action.type).toEqual(ACTION_TYPES.UPDATE_TAB_ROWS_DATA)
    expect(action.payload).toHaveProperty('data.AD_Tab-1');
    expect(action.payload).toMatchSnapshot();
  });
});

describe.skip('WindowActions thunks', () => {
  const fixturesData = fixtures.data1;
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);

  describe('init', () => {
    afterEach(() => {
      fetchMock.restore()
    });

    it(`dispatches 'initWindow' and 'initDataSuccess' action creators`, () => {
      const store = mockStore();
      const { windowType, docId } = fixturesData;

      nock(config.API_URL)
        .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
        .get(`rest/api/window/${windowType}/${docId}`)
        .reply(200, { data: { windowId: windowType } });

      const expectedActions = [
        { type: ACTION_TYPES.INIT_WINDOW },
        { type: ACTION_TYPES.INIT_DATA_SUCCESS, body: { windowId: windowType } }
      ]

      return store.dispatch(createWindow(windowType, docId, undefined, undefined, false)).then(() => {
        expect(store.getActions()).toEqual(expectedActions)
      });
    });
  });
})