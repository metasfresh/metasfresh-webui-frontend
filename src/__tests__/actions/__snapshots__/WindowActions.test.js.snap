// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowActions synchronous should return a UPDATE_TAB_ROWS_DATA action with correct payload 1`] = `
Object {
  "data": Object {
    "AD_Tab-1": Object {
      "changed": Object {
        "id": 1,
      },
      "removed": Object {},
    },
  },
  "scope": "master",
  "tabId": "AD_Tab-1",
}
`;
